activerecord/test/cases/connection_pool_test.rb

```diff
@@ -27,8 +31,13 @@ def setup
         end
       end
 
-      teardown do
+      def skip_fiber_testing
+        skip "Can't test isolation_level=fiber without a Ruby 3.1+ Fiber Scheduler" if @isolation_level == :fiber
+      end
+
+      def shared_teardown
         @pool.disconnect!
+        ActiveSupport::IsolatedExecutionState.isolation_level = @previous_isolation_level
       end
```

Comment:
Nitpick, rather than `shared_teardown / shared_setup` it should just be `teardown / setup` and then call `super` (except for teardown since you have no more code to add).